### PR TITLE
DOCS-Fix-landing-page-link

### PR DESCRIPTION
**Pull request by bmckercher123**
Update landing page component link to CE

Product Version(s):
CE 3.17.1

Issue:
n/a

Link to docs preview:

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

Merge checklist:

- [ ] Deploy preview inspected wherever changes were made 
- [ ] Build completed successfully
- [ ] Test have passed

### CODE REVIEW

File: static/_redirects [hunk 1]
```diff
@@ -52,7 +50,7 @@
 
 ## Reorganizing 'About' section for CE:
 #TODO CE 3.17.0: Uncomment next line
```
Comment (doucol) [inline static/_redirects]: remove the TODO here?

File: static/_redirects [hunk 2]
```diff
@@ -27,14 +26,13 @@
 /calico-cloud/ /calico-cloud/about 301
 
 #TODO CE 3.17.0: Replace next line with /calico-enterprise/ /calico-enterprise/latest/about 301
-/calico-enterprise/ /calico-enterprise/latest/about-calico-enterprise 301
+/calico-enterprise/ /calico-enterprise/latest/about 301
 /calico-enterprise/next/ /calico-enterprise/next/about 301
 #TODO CE 3.17.0: Replace next line with /calico-enterprise/latest /calico-enterprise/latest/about 301
```
Comment (doucol) [inline static/_redirects]: remove the TODO here?

File: static/_redirects [hunk 3]
```diff
@@ -27,14 +26,13 @@
 /calico-cloud/ /calico-cloud/about 301
 
 #TODO CE 3.17.0: Replace next line with /calico-enterprise/ /calico-enterprise/latest/about 301
```
Comment (doucol) [inline static/_redirects]: remove the TODO here?

**Review by doucol — CHANGES_REQUESTED** [review]

File: static/_redirects (hunk 3 above)
Comment (bmckercher123) [reply static/_redirects]: Fixed

File: static/_redirects (hunk 2 above)
Comment (bmckercher123) [reply static/_redirects]: Fixed

File: static/_redirects (hunk 1 above)
Comment (bmckercher123) [reply static/_redirects]: Fixed

**Review by doucol — APPROVED** [review]

**Pull request merged**